### PR TITLE
Add l1-output-null test

### PR DESCRIPTION
**Pull request by Frassle**
`null` is a valid value in the PCL model, and stacks should be able to return outputs of null directly, or with null nested inside other structures. Unfortunately that breaks a load of the rest of the model where we treat optional resource args which in the runtime language will just be null as not sent fields. So for now this test just checks that we can write null in PCL and return an array of null.

This also changes the l1-builtin-try test to return an array of null, rather than null directly to work around the above issue.

### CODE REVIEW

File: pkg/codegen/go/gen_program_expressions.go [hunk 1]
```diff
@@ -947,7 +947,7 @@ func (g *generator) argumentTypeName(destType model.Type, isInput bool) (result
 			}
 		}
 
-		if elmType != nil {
+		if elmType != nil && elmType != model.NoneType {
```
Comment (Frassle) [inline pkg/codegen/go/gen_program_expressions.go:950]: Without this go codegen would write a type `[]` that is a slice without the type inside it.

File: pkg/codegen/nodejs/gen_program_expressions.go [hunk 2]
```diff
@@ -703,7 +703,7 @@ func (g *generator) GenLiteralValueExpression(w io.Writer, expr *model.LiteralVa
 	case model.BoolType:
 		g.Fgenf(w, "%v", expr.Value.True())
 	case model.NoneType:
-		g.Fgen(w, "undefined")
```
Comment (Frassle) [inline pkg/codegen/nodejs/gen_program_expressions.go:706]: None in PCL is "null" not "undefined". undefined in Node really is "not present" and we elide them from stack outputs specifically. This _should_ eventually work to return null as a stack output though.

File: sdk/go/pulumi/rpc.go [hunk 3]
```diff
@@ -455,9 +455,7 @@ func marshalInputOptionsImpl(v interface{},
 				if err != nil {
 					return resource.PropertyValue{}, nil, err
 				}
-				if !e.IsNull() {
```
Comment (Frassle) [inline sdk/go/pulumi/rpc.go:458]: It's valid to have null values in arrays

File: cmd/pulumi-test-language/tests/l1_output_null.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+					snap *deploy.Snapshot, changes display.ResourceChanges,
+				) {
+					RequireStackResource(l, err, changes)
+					stack := RequireSingleResource(l, snap.Resources, "pulumi:pulumi:Stack")
+
+					outputs := stack.Outputs
+
+					assert.Len(l, outputs, 1, "expected 1 outputs")
+					// TODO: Ideally we'd allow for nulls in the output map and in nested maps, but to do that
+					// we need to work out how to handle optional fields in resource properties as well. Is it
```
Comment (brandonpollack23) [inline cmd/pulumi-test-language/tests/l1_output_null.go:39]: add issue number

File: cmd/pulumi-test-language/tests/testdata/l1-output-null/main.pp [hunk 5]
```diff
@@ -0,0 +1,13 @@
+//output "null" {
+//    value = null
+//}
```
[the remark this answers is not in the document]
Comment (Frassle) [reply cmd/pulumi-test-language/tests/testdata/l1-output-null/main.pp:3]: yeh I wanted these left in to show what _should_ work. I'll raise an issue and a comment here about it.

**Review by brandonpollack23 — APPROVED** [review]
